src/App.tsx

```diff
@@ -15,49 +16,117 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
-
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+ export const App: React.FC = () => {
+   const [sortType, setSortType] = useState(SortType.NONE);
+   const [isReversed, setIsReversed] = useState(false);
+
+   const sortByAlphabet = () => {
+     setSortType(SortType.ALPHABET);
+   };
+
+   const sortByLength = () => {
+     setSortType(SortType.LENGTH);
+   };
+
+   const reverse = () => {
+     setIsReversed(!isReversed);
+   };
+
+   const reset = () => {
+     setIsReversed(false);
+     setSortType(SortType.NONE);
+   };
+
+   const getReorderedGoods = (
+     goods: string[],
+     sort: SortType,
+     reversed: boolean,
+   ) => {
+     const visibleGoods = [...goods];
+
+     visibleGoods.sort((a, b) => {
+       switch (sort) {
+         case SortType.ALPHABET:
+           return a.localeCompare(b);
+
+         case SortType.LENGTH:
+           return a.length - b.length;
+
+         default:
+           return 0;
+       }
+     });
+
+     return reversed ? visibleGoods.reverse() : visibleGoods;
+   };
+
+   const sortedGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+
+   return (
+     <div className="section content">
+       <div className="buttons">
+         <button
+           type="button"
+      className={classNames(
+             'button',
+             'is-info',
+             { 'is-light': sortType !== SortType.ALPHABET },
+           )}
+           onClick={sortByAlphabet}
+         >
```

Comment:
pay attention to the code style and fix the whole file
```suggestion
         <button
           type="button"
           className={classNames(
             'button',
             'is-info',
             { 'is-light': sortType !== SortType.ALPHABET },
           )}
           onClick={sortByAlphabet}
         >
```